airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -104,7 +104,9 @@ def execute(self, context: 'Context'):
 
         :return: the id of the current glue job.
         """
-        if not self.script_location.startswith(self.s3_protocol):
+        if self.script_location is None:
+            s3_script_location = None
+        elif self.script_location.startswith(self.s3_protocol):
```

Comment:
```suggestion
        elif not self.script_location.startswith(self.s3_protocol):
```

Reply:
For additional context / reasoning, the logic is that if it's not an S3 file, it is assumed to be a local file, which is then uploaded to S3 first, and then referenced as "s3 script location".